### ISSUE TITLE
Semicolon rule marks colon in class arrow function as an error

### ISSUE THREAD

**Issue by ValeriiVasin** [palantir/tslint#2635]
### Bug Report

- __TSLint version__: 5.1.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class MyClass {
  handleLanguageChange = (language: string) =>
    this.switchLanguage(language);
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "no-internal-module": true,
        "typedef-whitespace": [
            true,
            {
              "call-signature": "nospace",
              "index-signature": "nospace",
              "parameter": "nospace",
              "property-declaration": "nospace",
              "variable-declaration": "nospace"
            }
        ],

        "curly": true,
        "forin": true,
        "no-bitwise": true,
        "no-arg": true,
        "no-conditional-assignment": true,
        "no-console": [
            true,
            "debug",
            "info",
            "time",
            "timeEnd",
            "trace",
            "log",
            "error"
        ],
        "no-construct": true,
        "no-debugger": true,
        "no-duplicate-variable": true,
        "no-empty": true,
        "no-eval": true,
        "no-invalid-this": true,
        "no-shadowed-variable": true,
        "no-string-literal": true,
        "no-switch-case-fall-through": true,
        "no-unused-expression": true,
        "no-unused-variable": true,
        "no-use-before-declare": true,
        "no-var-keyword": true,
        "radix": true,
        "switch-default": true,
        "triple-equals": true,
        "use-isnan": true,

        "eofline": true,
        "indent": [true, "spaces"],
        "linebreak-style": [true, "LF"],
        "max-line-length": [true, 120],
        "no-trailing-whitespace": true,
        "trailing-comma": [true, {
          "multiline": "always",
          "singleline": "never"
        }],

        "class-name": true,
        "comment-format": [
            true,
            "check-space"
        ],
        "interface-name": [true, "always-prefix"],
        "jsdoc-format": true,
        "no-consecutive-blank-lines": [true, 1],
        "one-line": [
          true,
          "check-catch",
          "check-finally",
          "check-else",
          "check-open-brace",
          "check-whitespace"
        ],
        "quotemark": [true, "single"],
        "semicolon": [true, "always"],
        "variable-name": [
            true,
            "ban-keywords"
        ],
        "whitespace": [
            true,
            "check-branch",
            "check-decl",
            "check-operator",
            "check-module",
            "check-separator",
            "check-type",
            "check-typecast"
        ]
    }
}
```

#### Actual behavior
```
ERROR: <FILE>: Unnecessary semicolon
```

#### Expected behavior
All should be fine
Also i would assume that it should add colon in such expression if missed.

**Comment by ajafff** [comment]
Try this:
```
"semicolon": [true, "always", "ignore-bound-class-methods"],
```

**Comment by ValeriiVasin** [comment]
@ajafff Awesome, thx!

Misunderstood the argument behavior... Would be awesome to see examples on each rule...
